lean4-infoview/src/infoview/info.tsx

```diff
@@ -0,0 +1,266 @@
+                onClick={e => { e.preventDefault(); onPin(pos); }}
+                title={isPinned ? 'unpin' : 'pin'} />
+            <a className={"link pointer mh2 dim codicon " + (isPaused ? "codicon-debug-continue" : "codicon-debug-pause")}
+               onClick={e => { e.preventDefault(); setPaused(!isPaused); }}
+               title={isPaused ? 'continue updating' : 'pause updating'} />
+            <a className="link pointer mh2 dim codicon codicon-refresh"
+               onClick={e => { e.preventDefault(); void triggerUpdate(); }}
+               title="update"/>
+        </span>
+    </summary>
+    );
+}
+
+interface InfoDisplayProps extends InfoPinnable {
+    pos: DocumentPosition;
+    status: InfoStatus;
+    messages: LeanDiagnostic[];
+    goal?: PlainGoal;
+    termGoal?: PlainTermGoal;
+    error?: string;
+    triggerUpdate: () => Promise<void>;
+}
+
+/** Displays goal state and messages. Can be paused. */
+export function InfoDisplay(props0: InfoDisplayProps) {
+    // Used to update the paused state once if a display update is triggered
+    const [shouldRefresh, setShouldRefresh] = React.useState<boolean>(false);
+    const [isPaused, setPaused, props, propsRef] = usePausableState(false, props0);
+    if (shouldRefresh) {
+        propsRef.current = props0;
+        setShouldRefresh(false);
+    }
+    const triggerDisplayUpdate = async () => {
+        await props0.triggerUpdate();
+        setShouldRefresh(true);
+    };
+
+    const {kind, pos, status, messages, goal, termGoal, error} = props;
+
+    const ec = React.useContext(EditorContext);
+    let copyGoalToComment: (() => void) | undefined = undefined;
+    if (goal) copyGoalToComment = () => { void ec.copyToComment(getGoals(goal).join('\n\n')); }
+    
+    // If we are the cursor infoview, then we should subscribe to
+    // some commands from the extension
+    const isCursor = kind === 'cursor';
+    useEvent(ec.events.requestedAction, act => {
+        if (!isCursor) return;
+        if (act.kind !== 'copyToComment') return;
+        copyGoalToComment && copyGoalToComment();
+    }, [goal]);
+    useEvent(ec.events.requestedAction, act => {
+        if (!isCursor) return;
+        if (act.kind !== 'togglePaused') return;
+        setPaused(isPaused => !isPaused);
+    });
+
+    const nothingToShow = !error && !goal && !termGoal && messages.length === 0;
+
+    return (
+    <Details initiallyOpen>
+        <InfoStatusBar {...props} triggerUpdate={triggerDisplayUpdate} isPaused={isPaused} setPaused={setPaused} copyGoalToComment={copyGoalToComment} />
+        <div className="ml1">
+            {status === 'error' && error &&
+                <div className="error">
+                    Error updating: {error}.
+                    <a className="link pointer dim" onClick={e => { e.preventDefault(); triggerDisplayUpdate(); }}>Try again.</a>
+                </div>}
+            {status !== 'error' && goal && 
```

Comment:
I didn't notice this before because github doesn't show the diff.  There is a good reason why we are using `display: none` in addition to conditional rendering: with `display: none`, the state of the details element is preserved if you move around the file (i.e. whether it's collapsed or not).  This allows you e.g. to disable the error message display, or the term goal display.

Reply:
Apologies about the diff, I was be careful to make the move commit contain Renames rather than Add/Delete, but alas it doesn't seem to work for all files. Yes, I see. I think this means we should put the stateful part (`Details`) in `Info` and the pure part in `Goal`. I will fix.